app/views/messages/_messages_for_library.html.erb

```diff
@@ -1,24 +1,48 @@
 <div class="messages">
-<% messages = @messages.select { |m| m.library == library_code && m.request_type == request_type } %>
-<% messages.each do |message| %>
+
+  <% messages = @messages.select { |m| m.library == library_code && m.request_type == request_type } %>
+
+  <% if library_code == 'SAL' %>
+    <% library_code = "SAL1&2" %>
+  <% elsif library_code == "SAL-NEWARK"%>
+    <% library_code = "SAL Newark" %>
+  <% end %>
+
+  <h2><%= request_type.titleize + " from " + library_code if messages.none? %></h2>
+  <%= link_to " Add message", new_message_path(library: library_code, request_type: request_type), class: 'btn btn-sm btn-default edit-message glyphicon glyphicon-pencil' if messages.none? %>
+
+  <% messages.each do |message| %>
+  <h2><%= request_type.titleize + " from " + library_code %></h2>
+  <% if message.text.length > 0 %>
+    <%= link_to [:edit, message], class: 'btn btn-sm btn-default edit-message' do %>
+      <i class="glyphicon glyphicon-pencil"> Edit message</i>
+    <% end %>
+  <% else %>
+    <%= link_to new_message_path(library: library_code, request_type: request_type), class: 'btn btn-sm btn-default edit-message' do %>
+      <i class="glyphicon glyphicon-pencil"> Add message</i>
```

Comment:
Move the text out of the `<i>` tag you won't get the weird spacing. This will be a bit more complicated with the `link_to` on line 12 but that's one of the reasons it's better to be consistent between them.